openldap-image/root/usr/local/bin/entrypoint.sh

```diff
@@ -11,56 +14,68 @@ ulimit -n 8192
+    echo "SLAPD_PASSWORD_FILE must be set so the initial admin account is created"
+    exit 1
+fi
 
-    if [ -z "$SLAPD_PASSWORD_FILE" ]; then
-        echo "SLAPD_PASSWORD_FILE must be set so the initial admin account is created"
-        exit 1
-    fi
+if [ -z "$SLAPD_DOMAIN" ]; then
+    echo "SLAPD_DOMAIN must be set with the initial domain name"
+    exit 1
+fi
 
-    if [ -z "$SLAPD_DOMAIN" ]; then
-        echo "SLAPD_DOMAIN must be set with the initial domain name"
-        exit 1
-    fi
+if [ -z "$SLAPD_ORGANIZATION" ]; then
+    SLAPD_ORGANIZATION="SUSE"
+fi
 
-    if [ -z "$SLAPD_ORGANIZATION" ]; then
-        SLAPD_ORGANIZATION="SUSE"
-    fi
+dc_string=""
 
-    dc_string=""
+IFS="."; declare -a dc_parts=($SLAPD_DOMAIN); unset IFS
 
-    IFS="."; declare -a dc_parts=($SLAPD_DOMAIN); unset IFS
+for dc_part in "${dc_parts[@]}"; do
+    dc_string="$dc_string,dc=$dc_part"
+done
 
-    for dc_part in "${dc_parts[@]}"; do
-        dc_string="$dc_string,dc=$dc_part"
-    done
+if [ -z "$SLAPD_ADMIN_USER" ] ; then
+    SLAPD_ADMIN_USER="cn=admin,${dc_string:1}"
+fi
 
-    if [ -z "$SLAPD_ADMIN_USER" ] ; then
-        SLAPD_ADMIN_USER="cn=admin,${dc_string:1}"
-    fi
+IFS=","; declare -a admin_user_parts=($SLAPD_ADMIN_USER); unset IFS
 
-    IFS=","; declare -a admin_user_parts=($SLAPD_ADMIN_USER); unset IFS
+base_string="BASE ${dc_string:1}"
+suffix_string="suffix \"${dc_string:1}\""
+rootdn_string="rootdn \"${SLAPD_ADMIN_USER}\""
 
-    base_string="BASE ${dc_string:1}"
-    suffix_string="suffix \"${dc_string:1}\""
-    rootdn_string="rootdn \"${SLAPD_ADMIN_USER}\""
+password=`cat $SLAPD_PASSWORD_FILE`
+password_hash=`slappasswd -s "${password}"`
+rootpwd_string="rootpw ${password_hash}"
 
-    password=`cat $SLAPD_PASSWORD_FILE`
-    password_hash=`slappasswd -s "${password}"`
-    rootpwd_string="rootpw ${password_hash}"
+echo "Copying configuration templates"
+cp -f $SLAPD_CONF_TEMPLATE $SLAPD_CONF
+cp -f $LDAP_CONF_TEMPLATE $LDAP_CONF
 
-    echo "Copying configuration templates"
-    cp -f $SLAPD_CONF_TEMPLATE /etc/openldap/slapd.conf
-    cp -f $LDAP_CONF_TEMPLATE /etc/openldap/ldap.conf
+echo "Setting up $SLAPD_CONF configuration file"
+sed -i "s|^suffix.*|${suffix_string}|g" $SLAPD_CONF
+sed -i "s|^rootdn.*|${rootdn_string}|g" $SLAPD_CONF
+sed -i "s|^rootpw.*|${rootpwd_string}|g" $SLAPD_CONF
 
-    echo "Setting up /etc/openldap/slapd.conf configuration file"
-    sed -i "s|^suffix.*|${suffix_string}|g" /etc/openldap/slapd.conf
-    sed -i "s|^rootdn.*|${rootdn_string}|g" /etc/openldap/slapd.conf
-    sed -i "s|^rootpw.*|${rootpwd_string}|g" /etc/openldap/slapd.conf
+if [ -n "$SLAPD_TLS_ENABLED" ]; then
+    echo "Configuring TLS support in $SLAPD_CONF"
+    cat >>$SLAPD_CONF <<EOF
+TLSProtocolMin 3.1
+TLSCipherSuite HIGH:!SSLv3:!SSLv2:!ADH
+TLSCACertificateFile /etc/openldap/pki/ca.crt
+TLSCertificateFile /etc/openldap/pki/openldap.crt
+TLSCertificateKeyFile /etc/openldap/pki/openldap.pem
+EOF
```

Comment:
This operation assumes the `$SLAPD_CONF` file is not mounted, something the entrypoint cannot assume I'd say. It will append on every startup, and if that file happens to be mounted (or mounted on the future), we'll be appending this over and over.

Reply:
If the `slapd.conf` file is mounted from the host, [this copy](https://github.com/kubic-project/container-images/pull/27/files#diff-a1b646525a06b7c42e451f854c7ab100R56) will fail.